### PR TITLE
Fix panic in election-provider offchain worker.

### PR DESCRIPTION
**Pull request by kianenigma**

### CODE REVIEW

**Review by bkchr — APPROVED** [review]

File: frame/election-provider-multi-phase/src/unsigned.rs [hunk 1]
```diff
@@ -437,6 +437,11 @@ impl<T: Config> Pallet<T> {
 		let mut high = assignments.len();
 		let mut low = 0;
 
+		// not much we can do if assignments are already empty.
+		if high == low {
```
Comment (coriolinus) [inline frame/election-provider-multi-phase/src/unsigned.rs:441]: Early return here won't hurt, but is insufficient to solve this. The problem occurs when, after binary search, `low == assignments.len()`. Line 449 then checks `assignments[..low+1]`, which generates the panic.

Suggested fix: in line 449:

```rust
if low <= assignments.len() && encoded_size_of(&assignments[..low + 1])? <= max_allowed_length {
```

**Review by thiolliere — APPROVED** [review]
I would be curious to understand how the number of assigment was empty. Did the election triggered on some unexpected context ?

Anyway this PR is correctly fixing a panic.

**Comment by kianenigma** [comment]
bot merge

**Comment by andresilva** [comment]
@thiolliere I was running a local testnet with 2 validators and this happened after a while, I didn't look into it any deeper.

**Comment by kianenigma** [comment]
my 2 cents: 

> from the PoV of the `election-provider-multi-phase`, if `DesiredTargets == 0` alongside a snapshot (which translates to `ValidatorCount == 0` in staking) then an empty assignment is totally valid and should still be processed without errors.
> I can see that it is a rare case, but the pallet should be able to deal with it.

**Pull request merged**